src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,45 @@
+package task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Deadline task in the chatbot system. A Deadline object
+ * contains a description, done status, and 'by' date.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
Perhaps the variable name can be changed to another name that has a clearer explanation?
```suggestion
    protected LocalDate dueByDate;
```